bench/bench.ml

```diff
@@ -0,0 +1,484 @@
+        benchmark = iter;
+        dependency = Some "replace_random";
+      };
+      {
+        name = "find_random_ro";
+        synopsis = "[RO] Find in random order";
+        readonly = true;
+        fresh = false;
+        benchmark = find_random;
+        dependency = Some "replace_random";
+      };
+      {
+        name = "find_random_rw";
+        synopsis = "[RW] Find in random order";
+        readonly = false;
+        fresh = false;
+        benchmark = find_random;
+        dependency = Some "replace_random";
+      };
+      {
+        name = "find_absent_ro";
+        synopsis = "[RO] Find absent values";
+        readonly = true;
+        fresh = false;
+        benchmark = find_absent;
+        dependency = Some "replace_random";
+      };
+      {
+        name = "find_absent_rw";
+        synopsis = "[RW] Find absent values";
+        readonly = false;
+        fresh = false;
+        benchmark = find_absent;
+        dependency = Some "replace_random";
+      };
+    ]
+end
+
+let list_benches () =
+  let pp_bench ppf b = Fmt.pf ppf "%s\t-- %s" b.Index.name b.synopsis in
+  Index.suite |> Fmt.(pr "%a" (list ~sep:Fmt.(const string "\n") pp_bench))
+
+let schedule p s =
+  let todos = List.map fst in
+  let init = ref (s |> List.map (fun b -> (p b.Index.name, b))) in
+  let apply_dep s =
+    let deps =
+      s
+      |> List.fold_left
+           (fun acc (todo, b) ->
+             if todo then
+               match b.Index.dependency with Some s -> s :: acc | None -> acc
+             else acc)
+           []
+    in
+    s |> List.map (fun (todo, b) -> (todo || List.mem b.Index.name deps, b))
+  in
+  let next = ref (apply_dep !init) in
+  while todos !init <> todos !next do
+    init := !next;
+    next := apply_dep !init
+  done;
+  let r = List.filter fst !init |> List.map snd in
+  r
+
+type config = {
+  key_size : int;
+  value_size : int;
+  nb_entries : int;
+  log_size : int;
+  seed : int;
+  with_metrics : bool;
+}
+[@@deriving yojson]
+
+let pp_config fmt config =
+  Format.fprintf fmt
+    "Key size: %d@\n\
+     Value size: %d@\n\
+     Number of bindings: %d@\n\
+     Log size: %d@\n\
+     Seed: %d@\n\
+     Metrics: %b" config.key_size config.value_size config.nb_entries
+    config.log_size config.seed config.with_metrics
+
+let cleanup root =
+  let files = [ "data"; "log"; "lock"; "log_async"; "merge" ] in
+  List.iter
+    (fun (b : Index.suite_elt) ->
+      let dir = root // b.name // "index" in
+      List.iter
+        (fun file ->
+          let file = dir // file in
+          if Sys.file_exists file then Unix.unlink file)
+        files)
+    Index.suite
+
+let init config =
+  Printexc.record_backtrace true;
+  Random.init config.seed;
+  if config.with_metrics then (
+    Metrics.enable_all ();
+    Metrics_gnuplot.set_reporter ();
+    Metrics_unix.monitor_gc 0.1 );
+  bindings_pool := make_bindings_pool config.nb_entries;
+  absent_bindings := make_bindings_pool config.nb_entries
```

Comment:
What do you think about generating this array only if the find absent benchmark is called? I think you can do that in the `run` function.

Reply:
If we do that, we will face the same issue you described above: whether an absent bench runs or not will change the results of the other benches.